workers/unity/Packages/com.improbable.gdk.buildsystem/Configuration/SpatialOSBuildConfigurationEditor.cs

```diff
@@ -212,237 +573,181 @@ private static TEnum EnumFlagsToggleField<TEnum>(TEnum source)
-                            {
-                                newBuildOptions ^= BuildOptions.Development;
-                            }
-
-                            prevValue = (newBuildOptions & BuildOptions.EnableHeadlessMode) != 0;
-                            newValue = EditorGUILayout.ToggleLeft("Headless Mode", prevValue);
-                            if (prevValue != newValue)
-                            {
-                                newBuildOptions ^= BuildOptions.EnableHeadlessMode;
-                            }
-                        }
-                    }
-                }
+            var problems = new List<Problem>();
+            var environmentConfiguration =
+                configurationForWorker.GetEnvironmentConfig(environment);
 
+            var buildOptions = environmentConfiguration.BuildOptions;
 
-                if ((newBuildOptions & BuildOptions.EnableHeadlessMode) != 0 &&
-                    (newBuildOptions & BuildOptions.Development) != 0)
+            if ((buildOptions & BuildOptions.EnableHeadlessMode) != 0 &&
+                (buildOptions & BuildOptions.Development) != 0)
+            {
+                problems.Add(new Problem
                 {
-                    EditorGUILayout.HelpBox(
-                        "You cannot have both EnableHeadlessMode and Development build enabled.\n" +
+                    Message = "You cannot have both EnableHeadlessMode and Development build enabled.\n" +
                         "This will crash the Unity Editor during the build.",
-                        MessageType.Error);
-                }
+                    Type = MessageType.Error
+                });
+            }
 
-                if ((newBuildOptions & BuildOptions.EnableHeadlessMode) != 0 &&
-                    (environmentConfiguration.BuildPlatforms & ~SpatialBuildPlatforms.Linux) != 0)
+            if ((buildOptions & BuildOptions.EnableHeadlessMode) != 0 &&
+                (environmentConfiguration.BuildPlatforms & ~SpatialBuildPlatforms.Linux) != 0)
+            {
+                problems.Add(new Problem
                 {
-                    EditorGUILayout.HelpBox(
+                    Message =
                         "EnableHeadlessMode is only available for Linux builds.",
-                        MessageType.Warning);
-                }
+                    Type = MessageType.Warning
+                });
+            }
 
-                if (EditorGUI.EndChangeCheck())
-                {
-                    EditorUtility.SetDirty(target);
-                    Undo.RecordObject(target, "Configure build options for worker");
+            var currentAdjustedPlatforms = environmentConfiguration.BuildPlatforms;
 
-                    environmentConfiguration.ShowBuildOptions = showBuildOptions;
-                    environmentConfiguration.BuildOptions = newBuildOptions;
-                }
+            if ((currentAdjustedPlatforms & SpatialBuildPlatforms.Current) != 0)
```

Comment:
Since we're now on a newer C#, can we use `currentAdjustedPlatforms.HasFlag` ?

Reply:
Done